From 1ad7ebecba68dc9cdb4d0087e2bd0e9db167d591 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Fri, 13 Feb 2015 14:40:32 +0530
Subject: [PATCH 085/148] eeprom/of: Add device tree bindings to at25.

Commit 658ec49417baceb094042f8181a86ce3e4df3ddf from
git://git.yoctoproject.org/linux-yocto-3.14

Add a MODULE_DEVICE_TABLE so the drivers can be automatically bound.

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Abhishek Paliwal <abhishek.paliwal@aricent.com>
---
 drivers/misc/eeprom/at25.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/misc/eeprom/at25.c b/drivers/misc/eeprom/at25.c
index 4f3bca1..377b35c 100644
--- a/drivers/misc/eeprom/at25.c
+++ b/drivers/misc/eeprom/at25.c
@@ -468,12 +468,20 @@ static const struct of_device_id at25_of_match[] = {
 };
 MODULE_DEVICE_TABLE(of, at25_of_match);
 
+static const struct spi_device_id at25_id[] = {
+	{"at25", 0},
+	{"m95256", 0},
+	{ }
+};
+MODULE_DEVICE_TABLE(spi, at25_id);
+
 static struct spi_driver at25_driver = {
 	.driver = {
 		.name		= "at25",
 		.owner		= THIS_MODULE,
 		.of_match_table = at25_of_match,
 	},
+	.id_table       = at25_id,
 	.probe		= at25_probe,
 	.remove		= at25_remove,
 };
-- 
1.8.2.1

